pulp_ansible/app/custom_fields.py

```diff
@@ -0,0 +1,209 @@
+import json
+
+from collections import OrderedDict
+
+from django.utils.translation import gettext_lazy as _
+from django.core.exceptions import BadRequest
+
+from drf_spectacular.utils import extend_schema_field
+
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+
+from pulpcore.plugin.models.role import Role
+
+from pulpcore.plugin.util import (
+    assign_role,
+    remove_role,
+    get_groups_with_perms_attached_roles,
+    get_perms_for_model,
+)
+from pulpcore.plugin.models import Group
+
+
+class RelatedFieldsBaseSerializer(serializers.Serializer):
```

Comment:
Is this a reinvention of drf_queryfields?

Reply:
I am not aware about drf_queryfields, but this is the pattern we use in other places in galaxy API

Reply:
So this is a bit of code copied from galaxy_ng?
Let me rephrase: Why is it called `include_related`? All I see is, it can select fields on the serializer to be included or not. And that is exactly what drf_queryfields (a pulp dependency for about forever) does.

Reply:
@mdellweg I talked to David and here is the explanation:

include_related controls what is included in the specific field that is based on RelatedFieldsSerializer, this is not to control the regular fields from the serializer, but to control which fields are included nested in a specialized field, the name is `include_related` because it is always data coming from related models.

So instead of always fetching the related models we do that only when is explicitly specified in the query param.